RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -0,0 +1,255 @@
+
+// k4FWCore
+#include "k4Interface/IGeoSvc.h"
+
+// edm4hep
+#include "edm4hep/ClusterCollection.h"
+
+// DD4hep
+#include "DD4hep/Detector.h"
+
+// ROOT
+#include "TLorentzVector.h"
+#include "TString.h"
+#include "TVector3.h"
+#include "TMath.h"
+
+DECLARE_COMPONENT(AugmentClustersFCCee)
+
+AugmentClustersFCCee::AugmentClustersFCCee(const std::string &name, ISvcLocator *svcLoc)
+    : Gaudi::Algorithm(name, svcLoc),
+      m_geoSvc("GeoSvc", "AugmentClustersFCCee")
+{
+  declareProperty("inClusters", m_inClusters, "Input clusters");
+  declareProperty("outClusters", m_outClusters, "Output clusters");
+}
+
+StatusCode AugmentClustersFCCee::initialize()
+{
+  {
+    StatusCode sc = Gaudi::Algorithm::initialize();
+    if (sc.isFailure())
+      return sc;
+  }
+
+  // check if readouts exist
+  for (size_t k = 0; k < m_readoutNames.size(); k++)
+  {
+    std::string readoutName = m_readoutNames[k];
+    if (m_geoSvc->getDetector()->readouts().find(readoutName) == m_geoSvc->getDetector()->readouts().end())
+    {
+      error() << "Readout <<" << readoutName << ">> does not exist." << endmsg;
+      return StatusCode::FAILURE;
+    }
+  }
+
+  // Check if readout related variables have the same size
+  if (m_systemIDs.size() != m_readoutNames.size())
+  {
+    error() << "Sizes of the systemIDs vector and readoutNames vector do not match, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_systemIDs.size() != m_numLayers.size())
+  {
+    error() << "Sizes of systemIDs vector and numLayers vector do not match, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_systemIDs.size() != m_layerFieldNames.size())
+  {
+    error() << "Sizes of systemIDs vector and layerFieldNames vector do not match, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_systemIDs.size() != m_detectorNames.size())
+  {
+    error() << "Sizes of systemIDs vector and detectorNames vector do not match, exiting!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // initialise the list of metadata for the clusters
+  // append to the metadata of the input clusters
+  std::string showerShapeDecorations = m_inShapeParameterHandle.get();
+  if (showerShapeDecorations != "")
+    showerShapeDecorations += ":";
+  for (size_t k = 0; k < m_detectorNames.size(); k++)
+  {
+    const char *detector = m_detectorNames[k].c_str();
+    for (unsigned layer = 0; layer < m_numLayers[k]; layer++)
+    {
+      if (layer > 0)
+        showerShapeDecorations += ":";
+      showerShapeDecorations += Form("f%s%d:theta%s%d:phi%s%d",
+        detector, layer, detector, layer, detector, layer);
+    }
+  }
+  m_showerShapeHandle.put(showerShapeDecorations.c_str());
+
+  return StatusCode::SUCCESS;
+}
+
+StatusCode AugmentClustersFCCee::finalize()
+{
+  return Gaudi::Algorithm::finalize();
+}
+
+StatusCode AugmentClustersFCCee::execute(const EventContext &evtCtx) const
+{
+  (void)evtCtx; // event context not used

```

Comment:
This is here just to avoid warning? Can you instead comment out the variable name?
```
StatusCode AugmentClustersFCCee::execute(const EventContext & /* evtCtx */) const
```

Reply:
Or probably better: `[[maybe_unused]]` (https://en.cppreference.com/w/cpp/language/attributes/maybe_unused)